bundles/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/NetatmoBinding.java

```diff
@@ -262,35 +262,34 @@ protected void execute() {
 						case STATIONNAME:
 							for (Device device : oauthCredentials.getStationsDataResponse
 									.getDevices()) {
-								if (stationPosition == null) {
-									DecimalType altitude = DecimalType.ZERO;
-									if (device.getAltitude() != null) {
-										altitude = new DecimalType(Math.round(unitSystem.
-												convertAltitude(device.getAltitude())));
-									}
-									stationPosition = new PointType(
-											new DecimalType(
-													new BigDecimal(device.getLatitude()).setScale(6, BigDecimal.ROUND_HALF_UP)),
-											new DecimalType(new BigDecimal(device.getLongitude()).setScale(6, BigDecimal.ROUND_HALF_UP)),
-											altitude);
-								}
 								if (device.getId().equals(deviceId)) {
+									if (stationPositions.get(device) == null) {
+										DecimalType altitude = DecimalType.ZERO;
+										if (device.getAltitude() != null) {
+											altitude = new DecimalType(Math.round(unitSystem.
+													convertAltitude(device.getAltitude())));
+										}
+										stationPositions.put(device, new PointType(
+												new DecimalType(new BigDecimal(device.getLatitude()).setScale(6, BigDecimal.ROUND_HALF_UP)),
+												new DecimalType(new BigDecimal(device.getLongitude()).setScale(6, BigDecimal.ROUND_HALF_UP)),
+												altitude));
```

Comment:
@jannegpriv, please do not make the change that @watou is suggesting. As I stated please leave as it was since it was this way before your PR.  @watou, it appears that Netatmo is sending the data in a way that introduces extra digits, and the changes that @jannegpriv are modifying more than the PR is adding in new functionality.